src/python/grpcio/tests/unit/framework/common/test_control.py

```diff
@@ -72,19 +73,31 @@ def control(self):
       if self._fail:
         raise Defect()
 
-      while self._paused:
+      while self._pause:
+        self._paused = True
+        self._condition.notify_all()
         self._condition.wait()
+      self._paused = False
 
   @contextlib.contextmanager
   def pause(self):
     """Pauses code under control while controlling code is in context."""
     with self._condition:
-      self._paused = True
+      self._pause = True
     yield
     with self._condition:
```

Comment:
After reading ```@contextlib.contextmanager```, what I said doesn't make sense...

What I intended is something along these lines:

```
@contextlib.contextmanager
with self._condition:
  self._pause = True
self._block_until_paused()
yield
self._pause = False
self._condition.notify_all()
```

Then anytime we have

```
with control.pause()
  #Do some stuff
````
We know that control is actually paused without needing to call  ```block_until_paused```